src/ds/ds-motion-common.h

```diff
@@ -0,0 +1,175 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "ds-device-common.h"
+#include <rsutils/string/from.h>
+#include "ds-calib-parsers.h"
+
+namespace librealsense
+{
+    // Enforce complile-time verification of all the assigned FPS profiles
+    enum class IMU_OUTPUT_DATA_RATES : uint16_t
+    {
+        IMU_FPS_63 = 63,
+        IMU_FPS_100 = 100,
+        IMU_FPS_200 = 200,
+        IMU_FPS_250 = 250,
+        IMU_FPS_400 = 400
+    };
+
+    using odr = IMU_OUTPUT_DATA_RATES;
+
+#ifdef _WIN32
+    static const std::string gyro_sensor_name = "HID Sensor Class Device: Gyroscope";
+    static const std::string accel_sensor_name = "HID Sensor Class Device: Accelerometer";
+    static const std::map<odr, uint16_t> hid_fps_translation =
+    {  //FPS   Value to send to the Driver
+        {odr::IMU_FPS_63,   1000},
+        {odr::IMU_FPS_100,  1000},
+        {odr::IMU_FPS_200,  500},
+        {odr::IMU_FPS_250,  400},
+        {odr::IMU_FPS_400,  250} };
+
+#else
+    static const std::string gyro_sensor_name = "gyro_3d";
+    static const std::string accel_sensor_name = "accel_3d";
+    static const std::map<IMU_OUTPUT_DATA_RATES, unsigned> hid_fps_translation =
+    {  //FPS   Value to send to the Driver
+        {odr::IMU_FPS_63,   1},
+        {odr::IMU_FPS_100,  1},
+        {odr::IMU_FPS_200,  2},
+        {odr::IMU_FPS_250,  3},
+        {odr::IMU_FPS_400,  4} };
+#endif
+
+    class auto_exposure_mechanism;
+    class fisheye_auto_exposure_roi_method : public region_of_interest_method
+    {
+    public:
+        fisheye_auto_exposure_roi_method(std::shared_ptr<auto_exposure_mechanism> auto_exposure)
+            : _auto_exposure(auto_exposure)
+        {}
+        void set(const region_of_interest& roi) override;
+        region_of_interest get() const override;
+
+    private:
+        std::shared_ptr<auto_exposure_mechanism> _auto_exposure;
+        region_of_interest _roi{};
+    };
+
+    class ds_fisheye_sensor : public synthetic_sensor, public video_sensor_interface, public roi_sensor_base, public fisheye_sensor
+    {
+    public:
+        explicit ds_fisheye_sensor(std::shared_ptr<sensor_base> sensor, device* owner);
+
+        rs2_intrinsics get_intrinsics(const stream_profile& profile) const override;
+        stream_profiles init_stream_profiles() override;
+        std::shared_ptr<uvc_sensor> get_raw_sensor();
+
+    private:
+        const std::vector<uint8_t>& get_fisheye_calibration_table() const;
+        std::shared_ptr<stream_interface> get_fisheye_stream() const;
+
+        device* _owner;
+    };
+
+    class ds_motion_sensor : public synthetic_sensor,
+                          public motion_sensor
+    {
+    public:
+        explicit ds_motion_sensor(std::string name,
```

Comment:
Do we use 2 Ctors?

Reply:
yes, one needs the sets for mapping of the profiles (uvc) and one does not (hid)